### PR TITLE
Shorten PR template checklist and remove verification workflow

### LINKED ISSUE #10970: [Discuss] Do not fail PRs on description non-compliance
From https://github.com/opensearch-project/OpenSearch/pull/10949#issuecomment-1781632280

Mechanisms like this are making PRs more burdensome. I suggest leaning out the GHA gate keeping to make it easier for PRs to merge, not more difficult. Processes like this are an oss turn off and increase contributor attrition.

### LINKED ISSUE #13965: [Feature Request] Remove or shorten the PR checklist
### Is your feature request related to a problem? Please describe

I personally find the PR checklist to be rote busy work that doesn't add a lot of value. Going entry by entry:

- **New functionality includes testing** - This is general software development best practice. Reviewers should be enforcing this. We have code coverage metrics published on each PR as well. I'm skeptical this checklist entry is helpful.
- **All tests pass** - PR workflows enforce this.
- **New functionality has been documented** - The `javadoc` targets enforce this as a part of the build if this refers to javadoc. If not, then this is duplicated by the "public documentation" entry below.
- **New functionality has javadoc added** - Same as above
- **API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)** - I could be convinced this one should stay, maybe.
- **Failing checks are inspected and point to the corresponding known issue(s)** - The failing checks post a comment linking to the flaky test guide telling you want to do.
- **Commits are signed per the DCO using --signoff** - The DCO workflow enforces this
- **Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))** - The changelog workflow enforces this.
- **Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)** - I could be convinced this one should stay, maybe.

### Describe the solution you'd like

Remove all checklist items except the "API Changes" and "Public Documentation" entries. Maybe consider removing the entire checklist if we don't have evidence the remaining two entries are effective.

### Related component

Other

### PR DESCRIPTION
**Pull request by andrross**
This commit removes items from the checklist that were redundant in order to reduce the work required to submit a PR.

### Related Issues
Resolves #13965
Resolves #10970

### Check List
- [x] Functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/pull-request-checks.yml [hunk 1]
```diff
@@ -21,9 +21,4 @@ jobs:
         if: github.event.pull_request.user.login != 'dependabot[bot]'
         with:
           checklist-items: |
-            New functionality includes testing.
-            All tests pass
-            New functionality has been documented.
-            New functionality has javadoc added
-            Commits are signed per the DCO using --signoff
-            Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
+            Functionality includes testing.
```
Comment (andrross) [inline .github/workflows/pull-request-checks.yml:24]: The "API changes" and "Public documentation" lines were not present in this verification script. I'm not sure if that was intentional (they are not always needed), so I didn't add them here.

**Review by kotwanikunal — APPROVED** [review]

**Review by reta — APPROVED** [review]

**Review by mch2 — APPROVED** [review]

**Comment by reta** [comment]
@peternied are you OK with this checklist? thank you

File: .github/workflows/pull-request-checks.yml (hunk 1 above)
Comment (peternied) [inline .github/workflows/pull-request-checks.yml:24]: What do you think about removing this check entirely? Seems to me that If every item isn't important to require an 'affirmative check', then none of them are - it isn't worth the cognitive load on contributors. 

I know disabling this check will be popular with based on [1] previous discussions.

- [1] https://github.com/opensearch-project/OpenSearch/issues/10970

**Review by peternied — APPROVED** [review]
@andrross Thanks for making these changes, lets see how this works out.

File: .github/workflows/pull-request-checks.yml (hunk 1 above)
Comment (andrross) [reply .github/workflows/pull-request-checks.yml:24]: Sounds good, I'll remove and update this PR.

**Review by peternied — APPROVED** [review]

**Pull request merged**